src/overlays/actors/ovl_Boss_02/z_boss_02.c

```diff
@@ -97,186 +151,2144 @@ const ActorInit Boss_02_InitVars = {
+                                                    0.0f, 60.0f, 0.0f, 0, 0, 0, 1);
+        Actor_Spawn(&globalCtx->actorCtx, globalCtx, ACTOR_ITEM_B_HEART, 0.0f, 30.0f, -150.0f, 0, 1, 0, 0);
+    }
+
+    this->actor.targetMode = 10;
+    this->unk_1D3C.z = this->unk_1D3C.x = 0.0f;
+    this->unk_1D3C.y = 1.0f;
+    if (this->actor.params == 200) {
+        D_809E042C = this;
+        globalCtx->specialEffects = (void*)D_809E0438;
+        this->actor.update = func_809DC78C;
+        this->actor.draw = func_809DD0A8;
+        this->actor.flags &= ~1;
+        this->unk_1D70 = 0.00999999977648f;
+        if ((KREG(64) != 0) || (gSaveContext.eventInf[5] & 0x20) || (D_809E0434 != NULL)) {
+            this->unk_1D20 = 0;
+            D_809E0430 = KREG(15) + 20;
+        } else {
+            this->unk_1D20 = 1;
+        }
+        XREG(41) = KREG(14) + 20;
+        this->unk_01AC = 1.0f;
+        Actor_SpawnAsChild(&globalCtx->actorCtx, &this->actor, globalCtx, ACTOR_EN_TANRON5, 0.0f, 1000.0f, 0.0f, 0, 0,
+                           0, 0);
+    } else if (this->actor.params == 100) {
+        this->actor.update = func_809DC218;
+        this->actor.draw = NULL;
+        this->actor.hintId = 0x2E;
+    } else {
+        if (this->actor.params != 35) {
+            this->actor.params = 0;
+            Actor_Spawn(&globalCtx->actorCtx, globalCtx, ACTOR_BOSS_02, 0.0f, 0.0f, 0.0f, 0, 0, 0, 200);
+            D_809E0424 = this;
+            D_809E0428 = (Boss02*)Actor_Spawn(&globalCtx->actorCtx, globalCtx, ACTOR_BOSS_02, this->actor.world.pos.x,
+                                              this->actor.world.pos.y, this->actor.world.pos.z, this->actor.world.rot.x,
+                                              this->actor.world.rot.y, this->actor.world.rot.z, 35);
+            D_809E0424->actor.colChkInfo.damageTable = &sDamageTable2;
+            D_809E0428->actor.colChkInfo.damageTable = &sDamageTable1;
+            D_809E0424->unk_1674 = D_809E0428;
+            D_809E0428->unk_1674 = D_809E0424;
+        }
+
+        this->actor.colChkInfo.mass = MASS_HEAVY;
+        this->actor.colChkInfo.health = 20;
+        Actor_SetScale(&this->actor, 0.01f);
+        SkelAnime_Init(globalCtx, &this->skelAnime, &D_06009B10, &D_06009C78, this->jointTable, this->morphTable, 13);
+        Collider_InitAndSetJntSph(globalCtx, &this->colliderSphere1, &this->actor, &sJntSphInit1,
+                                  this->colliderSphere1Elements);
+        Collider_InitAndSetJntSph(globalCtx, &this->colliderSphere2, &this->actor, &sJntSphInit2,
+                                  this->colliderSphere2Elements);
+        Collider_InitAndSetCylinder(globalCtx, &this->colliderCylinder, &this->actor, &sCylinderInit);
+
+        if (D_809E0434 != NULL) {
+            func_809DAA74(this, globalCtx);
+        } else {
+            func_809DAAA8(this, globalCtx);
+        }
+
+        for (i = 0; i < 200; i++) {
```

Comment:
```suggestion
        for (i = 0; i < ARRAY_COUNT(this->unk_01BC); i++) {
```

Reply:
Can't be done, I don't ahve access to `this`. I wanted to use this, but it's not passed into the function, so it'd have to be a define at best.